### ISSUE TITLE
ProgressStep missing label

### ISSUE THREAD

**Issue by nictownsend** [carbon-design-system/carbon#5113]
ProgressStep is not setting/using a <label> for the <div role="button">

## Environment

Chrome Mac OS X
DAP chrome plugin
Jan2020 archive, IBM Accessibility January 2020 ruleset

## Detailed description

> What version of the Carbon Design System are you using?

Carbon Components React 7.9.2

> What WCAG 2.1 checkpoint does the issue violate?

1.3.1 & 4.1.2

## Steps to reproduce the issue

https://codesandbox.io/s/serene-galois-tpk91

## Additional information

![image](https://user-images.githubusercontent.com/47386186/72815456-943f7300-3c5e-11ea-99ef-3cf5c5ea2dcd.png)

**Comment by joshblack** [comment]
Hi there! :wave: I'm unable to reproduce this violation with the given codesandbox and running the IBM Accessibility WCAG 2.1 September 2019 ruleset, could you provide additional steps to help reproduce?

**Comment by nictownsend** [comment]
Hi Josh, this is using Jan2020 archive, IBM Accessibility January 2020 ruleset

**Comment by joshblack** [comment]
It seems like the node itself does have content that will be used as a label, this could be similar to #5112 in that it's a false positive with DAP. When looking at it, it does seem like the content is used as the label for the button when a screen reader focuses on it.

**Comment by tw15egan** [comment]
Does not seem to have any DAP violations when running this locally...

<img width="1272" alt="Screen Shot 2020-02-25 at 1 27 13 PM" src="https://user-images.githubusercontent.com/11928039/75247604-9d2de200-57d2-11ea-92c6-3ad4c5d77f67.png">

**Comment by tw15egan** [comment]
Possibly due to this PR (https://github.com/carbon-design-system/carbon/pull/5125)  and switching the use of `div` to `button`

**Comment by tw15egan** [comment]
Going to close this as resolved and will be fixed when `10.10.0` is released
